proc/threads.c

```diff
@@ -1246,29 +1246,57 @@ void proc_threadBroadcastYield(thread_t **queue)
 }
 
 
-int proc_join(time_t timeout)
+int proc_join(int tid, time_t timeout)
 {
-	int err;
-	thread_t *ghost;
-	spinlock_ctx_t sc;
+	int err = EOK, found = 0, id = 0;
 	process_t *process = proc_current()->process;
+	thread_t *ghost = process->ghosts, *firstGhost = process->ghosts, *lastChecked = NULL;
```

Comment:
I think this read should happen under spinlock (happily writes to `ghosts` field is done under `threads_common.spinlock` too)